analyzer-core/src/main/java/ffb/analyzer/core/pubsub/Publisher.java

```diff
@@ -0,0 +1,79 @@
+package ffb.analyzer.core.pubsub;
+
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.Map;
+import java.util.Set;
+
+public class Publisher {
```

Comment:
After looking at your diagram, I'm not even sure we need a publisher. `PriorityQueueHandler` can handle notifying the subscribers once the task is complete.

If we really want a `Publisher` class, it doesn't need to know what event it's publishing. It should just take a result and `Subscriber`, and notify the subscriber of the result.

Reply:
Yeah, I may have jumped the gun on the pub-sub portion. I think what I have in place now handles the notification without it. Unless you feel strongly, I think it would still be worth keeping around in case we need it in the future. If you'd prefer it to not clutter up the project, I can store it separately on my computer.